hazelcast/src/main/java/com/hazelcast/datalink/impl/hazelcastdatalink/HazelcastDataLinkClientConfigBuilder.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.datalink.impl.hazelcastdatalink;
+
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.config.DataLinkConfig;
+import com.hazelcast.core.HazelcastException;
+import com.hazelcast.datalink.HazelcastDataLink;
+import com.hazelcast.internal.util.StringUtil;
+import com.hazelcast.jet.impl.util.ImdgUtil;
+
+public class HazelcastDataLinkClientConfigBuilder {
```

Comment:
I think we can merge the Builder, Loader, and Validator into a single class.

Reply:
I think that would violate the single responsibility principle.  Each class has a single role now